.gitignore

```diff
@@ -1 +1,84 @@
-*.sublime-workspace
\ No newline at end of file
+## PureBasic
+############
+*.pb.cfg
+*.pbi.cfg
+*.pbp
+project.cfg
+
+
+## Sublime Text 4
+#################
+*.sublime-project
+*.sublime-workspace
+
+
+## OS Specific
+##############
+## Created by https://www.gitignore.io/api/linux,macos,windows
+
+### Linux ###
+*~
+
+# temporary files which can be created if a process still has a handle open of a deleted file
+.fuse_hidden*
+
+# KDE directory preferences
+.directory
+
+# Linux trash folder which might appear on any partition or disk
+.Trash-*
+
+# .nfs files are created when an open file is removed but is still being accessed
+.nfs*
+
+### macOS ###
+
+# General
+.DS_Store
+.AppleDouble
+.LSOverride
+
+# Icon must end with two \r
+Icon
+
+# Thumbnails
+._*
+
+# Files that might appear in the root of a volume
+.DocumentRevisions-V100
+.fseventsd
+.Spotlight-V100
+.TemporaryItems
+.Trashes
+.VolumeIcon.icns
+.com.apple.timemachine.donotpresent
+
+# Directories potentially created on remote AFP share
+.AppleDB
+.AppleDesktop
+Network Trash Folder
+Temporary Items
+.apdisk
+
+### Windows ###
+
+# Windows thumbnail cache files
+Thumbs.db
+ehthumbs.db
+ehthumbs_vista.db
+
+# Dump file
+*.stackdump
+
+# Folder config file
+[Dd]esktop.ini
+
+# Recycle Bin used on file shares
+$RECYCLE.BIN/
+
+# Windows Installer files
```

Comment:
At one hand I feel unsure about the need of some exclusions, especially these. On the other hand I know that they do no harm, too. Such binaries probably are better kept in the global ignore file, I would guess.

Reply:
When it comes to repository configurations I prefer the "set it and forge it" approach, i.e. covering all general occurrences and common repository patterns, so that if new contents are submitted the settings are all in place.

E.g. I've added all the known ST file extensions to the config, even if we don't use all of them right now, so if we add new files we don't have to tweak the configuration every time.

> Such binaries probably are better kept in the global ignore file, I would guess.

They are added to prevent unwanted files in third party pull requests, especially in case of Git newbies. We can't make any assumption on the Git and machine settings of contributors.

These patterns were taken from the gitignore project which advises best practices for cross-platform projects, covering all the common ignore patterns for the major OSs, including old versions of Windows. The idea is to prevent setting files from ending up in commits, especially in third party contributions. 

That's why I think it's best to have them in the repository configuration, to be on the safe side — they are harmless, but they can prevent garbage ending up in contributions.

Because this is a repository related to PureBasic, I'm adding extra comments and taking extra precautions, because for the PB community using Git is something relatively new, prompted by the IDE source having been released on GitHub.

In the official PB IDE repository, we've seen a lot of pull requests with whitespace noise, system files, etc., and we had to take extra measures to enforce CI validation to keep contributions clean.